src/joyaccelerometersensor.cpp

```diff
@@ -187,34 +187,28 @@ JoySensorDirection JoyAccelerometerSensor::calculateSensorDirection()
     bool inPitch = pitch_abs < range;
     bool inRoll = roll_abs < range;
 
-    if (inPitch && !inRoll)
+    if (!inPitch && !inRoll)
     {
         if (roll > 0)
-            return SENSOR_LEFT;
-        else
-            return SENSOR_RIGHT;
-    } else if (!inPitch && inRoll)
-    {
-        if (pitch > 0)
-            return SENSOR_UP;
-        else
-            return SENSOR_DOWN;
-    } else // in both or in none
-    {
-        if (pitch > 0)
         {
-            if (roll > 0)
-                return SENSOR_LEFT_UP;
-            else
-                return SENSOR_RIGHT_UP;
+            return pitch > 0 ? SENSOR_LEFT_UP : SENSOR_LEFT_DOWN;
         } else
         {
-            if (roll > 0)
-                return SENSOR_LEFT_DOWN;
-            else
-                return SENSOR_RIGHT_DOWN;
+            return pitch > 0 ? SENSOR_RIGHT_UP : SENSOR_RIGHT_DOWN;
         }
     }
+
+    if (inPitch && !inRoll)
+    {
+        return roll > 0 ? SENSOR_LEFT : SENSOR_RIGHT;
+    }
+
+    if (!inPitch && inRoll)
+    {
+        return pitch > 0 ? SENSOR_UP : SENSOR_DOWN;
+    }
+
```

Comment:
Could you reorganize these if statements in a bit more nested manner?

Instead of 

```cpp
if (a && b){}

if(a && !b) {}
```

there should bemething like:

```cpp
if (a)
{
  if(b){}
  else {}
}
```